### PR TITLE
test(show-edit-pod): add cypress tests

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
Refactor for show-edit-pod component to use new cypress-component-react framework for testing.
I have added test.js and deleted steps.js and feature file.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [x] UI Tests GitHub check reviewed if required

### Testing instructions

- [x]  Run npx cypress open --component to check if there is newly added test.file
- [x]  Check if the show-edit-pod.test.js file passed
- [x]  Compare all tests were moved/refactored from .feature file -> .test.js file
- [x] Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: cypress/locators/show-edit-pod/index.js [hunk 1]
```diff
@@ -1,22 +1,36 @@
 import {
-  SHOW_EDIT_POD_TITLE,
-  SHOW_EDIT_SAVE_BUTTON,
-  SHOW_EDIT_CANCEL_BUTTON,
-  SHOW_EDIT_DELETE_BUTTON,
-  SHOW_EDIT_POD_EDIT,
-  SHOW_EDIT_POD_CONTENT_BOX,
+  SHOW_EDIT_POD,
```
Comment (sianford) [inline cypress/locators/show-edit-pod/index.js:2]: Why have you changed the format of the naming convention from COMPONENT_ELEMENT to ELEMENT_COMPONENT?

File: cypress/locators/show-edit-pod/index.js [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
-export const showEditPodEdit = () => cy.get(SHOW_EDIT_POD_EDIT);
-export const showEditPodCollapsibleInnerContent = (element, index) =>
-  cy
-    .get(SHOW_EDIT_POD_CONTENT_BOX)
-    .find("div")
-    .eq(1)
-    .find(`[data-element=${element}]`)
-    .eq(index);
+export const showEditPod = () => cy.get(SHOW_EDIT_POD).find("div");
+export const ContentShowEditPod = () => cy.get(CONTENT_SHOW_EDIT_POD);
```
Comment (sianford) [inline cypress/locators/show-edit-pod/index.js:19]: Why have you changed the naming convention from lower camel case (showEditPod) to upper camel case (ContentShowEditPod) and switched the componentElement to ElementComponent?

File: cypress/locators/show-edit-pod/locators.js [hunk 3]
```diff
@@ -1,9 +1,16 @@
 // component preview locators
-export const SHOW_EDIT_POD_CONTENT_BOX = '[data-element="content"]';
-export const SHOW_EDIT_POD_CONTENT_ELEMENTS = '[data-component="content"]';
-
-export const SHOW_EDIT_POD_TITLE = "h4[data-element=title]";
-export const SHOW_EDIT_SAVE_BUTTON = 'button[data-element="submit-button"]';
-export const SHOW_EDIT_CANCEL_BUTTON = 'button[data-element="cancel-button"]';
-export const SHOW_EDIT_DELETE_BUTTON = "[data-element=delete-button]";
-export const SHOW_EDIT_POD_EDIT = '[data-element="edit"]';
+export const SHOW_EDIT_POD = '[data-component="pod"]';
```
Comment (sianford) [inline cypress/locators/show-edit-pod/locators.js:2]: Why have you changed the format of the naming convention from COMPONENT_ELEMENT to ELEMENT_COMPONENT?

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 4]
```diff
@@ -0,0 +1,323 @@
+import * as React from "react";
+import ShowEditPod from "./show-edit-pod.component";
+import Textbox from "../textbox/textbox.component";
+import Content from "../content/content.component";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+import {
+  showEditPod,
+  ContentShowEditPod,
```
Comment (sianford) [inline src/components/show-edit-pod/show-edit-pod.test.js:9]: As above, the format of the name should be lower camel case.

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  DeleteButtonShowEditPod,
+  TitleShowEditPod,
+  EditContainerShowEditPod,
+  undoShowEditPod,
+  transitionNameShowEditPod,
+} from "../../../cypress/locators/show-edit-pod";
+
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+
+const ShowEditPodComponent = ({ ...props }) => {
```
Comment (sianford) [inline src/components/show-edit-pod/show-edit-pod.test.js:26]: As above, the format of the name should be lower camel case.

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+      [
+        false,
+        "rgb(0, 126, 69) none repeat scroll 0% 0% / auto padding-box border-box",
+      ],
+      [
+        true,
+        "rgb(230, 235, 237) none repeat scroll 0% 0% / auto padding-box border-box",
+      ],
+    ])(
+      "should check saveText as %s for ShowEditPod component",
```
Comment (sianford) [inline src/components/show-edit-pod/show-edit-pod.test.js:227]: I think the title of this test is misleading. It should be changed to something like "should check saving state is %s for ShowEditPod component".

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+    it.each([
+      [false, "rgba(0, 0, 0, 0)"],
+      [true, "rgb(230, 235, 237)"],
+    ])(
+      "should check when softDelete is %s for ShowEditPod component",
+      (boolVal, backgroundColor) => {
+        CypressMountWithProviders(
+          <ShowEditPodComponent softDelete={boolVal} />
+        );
+        showEditPod().should("have.css", "background-color", backgroundColor);
```
Comment (sianford) [inline src/components/show-edit-pod/show-edit-pod.test.js:131]: Part of the softDelete state includes the button in the pod being an 'Undo' button so you should assert that this is the case too.

File: cypress/locators/show-edit-pod/index.js (hunk 2 above)
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:19]:
```suggestion
export const contentShowEditPod = () => cy.get(CONTENT_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 8]
```diff
@@ -1,22 +1,36 @@
...
+export const ClassShowEditPod = () => cy.get(SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:20]: Why it is only `class`?
```suggestion
export const classShowEditPod = () => cy.get(SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 9]
```diff
@@ -1,22 +1,36 @@
...
+export const ContentFormShowEditPod = () => cy.get(CONTENT_FORM_SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:21]:
```suggestion
export const contentFormShowEditPod = () => cy.get(CONTENT_FORM_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 10]
```diff
@@ -1,22 +1,36 @@
...
+export const HideDeleteButtonShowEditPod = () =>
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:22]:
```suggestion
export const hideDeleteButtonShowEditPod = () =>
```

File: cypress/locators/show-edit-pod/index.js [hunk 11]
```diff
@@ -1,22 +1,36 @@
...
+  cy.get(HIDE_DELETE_BUTTON_SHOW_EDIT_POD);
+export const editFieldsShowEditPod = () => cy.get(EDIT_FIELDS_SHOW_EDIT_POD);
+export const FormFooterShowEditPod = () => cy.get(FORM_FOOTER_SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:25]:
```suggestion
export const formFooterShowEditPod = () => cy.get(FORM_FOOTER_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 12]
```diff
@@ -1,22 +1,36 @@
...
+export const CancelButtonShowEditPod = () =>
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:26]:
```suggestion
export const cancelButtonShowEditPod = () =>
```

File: cypress/locators/show-edit-pod/index.js [hunk 13]
```diff
@@ -1,22 +1,36 @@
...
+  cy.get(CANCEL_BUTTON_SHOW_EDIT_POD);
+export const SaveButtonShowEditPod = () => cy.get(SAVE_BUTTON_SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:28]:
```suggestion
export const saveButtonShowEditPod = () => cy.get(SAVE_BUTTON_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 14]
```diff
@@ -1,22 +1,36 @@
...
+export const DeleteButtonShowEditPod = () =>
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:29]:
```suggestion
export const deleteButtonShowEditPod = () =>
```

File: cypress/locators/show-edit-pod/index.js [hunk 15]
```diff
@@ -1,22 +1,36 @@
...
+  cy.get(DELETE_BUTTON_SHOW_EDIT_POD);
+export const TitleShowEditPod = () => cy.get(TITLE_SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:31]:
```suggestion
export const titleShowEditPod = () => cy.get(TITLE_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/index.js [hunk 16]
```diff
@@ -1,22 +1,36 @@
...
+export const EditContainerShowEditPod = () =>
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:32]:
```suggestion
export const editContainerShowEditPod = () =>
```

File: cypress/locators/show-edit-pod/index.js [hunk 17]
```diff
@@ -1,22 +1,36 @@
...
+  cy.get(EDIT_CONTAINER_SHOW_EDIT_POD);
+export const undoShowEditPod = () => cy.get(UNDO_BUTTON_SHOW_EDIT_POD);
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/index.js:34]:
```suggestion
export const undoButtonShowEditPod = () => cy.get(UNDO_BUTTON_SHOW_EDIT_POD);
```

File: cypress/locators/show-edit-pod/locators.js (hunk 3 above)
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/locators.js:2]:
```suggestion
export const POD = '[data-component="pod"]';
```

File: cypress/locators/show-edit-pod/locators.js [hunk 18]
```diff
@@ -1,9 +1,16 @@
...
+export const CONTENT_SHOW_EDIT_POD = '[data-component="content"]';
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/locators.js:3]: But I think there is such locator in `content`

```suggestion
export const CONTENT = '[data-component="content"]';
```

File: cypress/locators/show-edit-pod/locators.js [hunk 19]
```diff
@@ -1,9 +1,16 @@
...
+export const CONTENT_FORM_SHOW_EDIT_POD = '[data-element="form-content"]';
+export const HIDE_DELETE_BUTTON_SHOW_EDIT_POD = "button[data-element=delete]";
+export const EDIT_FIELDS_SHOW_EDIT_POD = '[data-element="input"]';
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/locators.js:6]: there is such locator in `common` locators

File: cypress/locators/show-edit-pod/locators.js [hunk 20]
```diff
@@ -1,9 +1,16 @@
...
+export const FORM_FOOTER_SHOW_EDIT_POD = '[data-element="form-footer"]';
```
Comment (DlgSHi) [inline cypress/locators/show-edit-pod/locators.js:7]: there is such locator in `common` locators

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 21]
```diff
@@ -0,0 +1,323 @@
...
+      saveText="Save"
+      deleteText="Delete"
+      cancelText="Cancel"
+      editFields={<Textbox label="Field" labelInline labelAlign="right" />}
+      {...props}
+    />
+  );
+};
+
+const UndoShowEditPodComponent = ({ ...props }) => {
```
Comment (DlgSHi) [inline src/components/show-edit-pod/show-edit-pod.test.js:41]: is there a possibility to combine this one with the above custom component?

File: src/components/show-edit-pod/show-edit-pod.test.js [hunk 22]
```diff
@@ -0,0 +1,323 @@
...
+        "rgb(230, 235, 237) none repeat scroll 0% 0% / auto padding-box border-box",
+      ],
+    ])(
+      "should check saveText as %s for ShowEditPod component",
+      (boolVal, color) => {
+        CypressMountWithProviders(
+          <ShowEditPodComponent editing saving={boolVal} />
+        );
+
+        SaveButtonShowEditPod().should("have.css", "background", color);
```
Comment (DlgSHi) [inline src/components/show-edit-pod/show-edit-pod.test.js:233]: if you will specify `background-***` we will be able to assert only colors

File: cypress/locators/show-edit-pod/locators.js (hunk 20 above)
Comment (divyajindel) [reply cypress/locators/show-edit-pod/locators.js:7]: we don't have "form-footer" in common, I found content and input.

File: src/components/show-edit-pod/show-edit-pod.test.js (hunk 21 above)
Comment (divyajindel) [reply src/components/show-edit-pod/show-edit-pod.test.js:41]: We don't want to use undo button for other tests as it's not common so I've created a new Component and position is same for edit and undo button so its better to keep it separate.

File: cypress/locators/show-edit-pod/index.js [hunk 23]
```diff
@@ -1,22 +1,32 @@
 import {
-  SHOW_EDIT_POD_TITLE,
-  SHOW_EDIT_SAVE_BUTTON,
-  SHOW_EDIT_CANCEL_BUTTON,
-  SHOW_EDIT_DELETE_BUTTON,
-  SHOW_EDIT_POD_EDIT,
-  SHOW_EDIT_POD_CONTENT_BOX,
+  POD,
+  CONTENT_FORM_SHOW_EDIT_POD,
```
Comment (sianford) [inline cypress/locators/show-edit-pod/index.js:3]: Why did you switch around the format of these locators from, for example, SHOW_EDIT_POD_SAVE_BUTTON to SAVE_BUTTON_SHOW_EDIT_POD when all other components are the original format of COMPONENT_ELEMENT as far as possible? Similarly for the corresponding indexes from showEditSaveButton to saveButtonShowEditPod? The format should be as uniform as possible.

Comment (divyajindel) [reply cypress/locators/show-edit-pod/index.js]: I have changed the order now.

**Review by sianford — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**